### PR TITLE
[TypeSpec Validation] exclude trailing slash when calculating folder depth

### LINKED ISSUE #26549: [TypeSpec Validation] Incorrectly includes trailing slash when calculating folder depth
```
Executing rule: FolderStructure
folder: specification/cognitiveservices/HealthInsights/HealthInsights.OncoPhenotype/
Rule FolderStructure failed
Please limit TypeSpec folder depth to 3 levels or less
```

### PR DESCRIPTION
**Pull request by ckairen**
closes #26549

### CODE REVIEW

File: eng/tools/typespec-validation/src/rules/folder-structure.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export class FolderStructureRule implements Rule {
     });
 
     // Verify top level folder is lower case
-    let folderStruct = folder.split("/");
+    let folderStruct = folder.split("/").filter(Boolean);
```
Comment (konrad-jamrozik) [inline eng/tools/typespec-validation/src/rules/folder-structure.ts:34]: Can you add a comment explaining what this does? I see like no connection between adding `filter(Boolean)` and making trailing slash disappear.

Comment (konrad-jamrozik) [reply eng/tools/typespec-validation/src/rules/folder-structure.ts:34]: Or even you could introduce inline/local function named `excludeTrailingSlash` that does it. "self-documenting code" for the win ;)

File: eng/tools/typespec-validation/src/rules/folder-structure.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export class FolderStructureRule implements Rule {
     });
 
     // Verify top level folder is lower case
```
Comment (ckairen) [inline eng/tools/typespec-validation/src/rules/folder-structure.ts:33]:
```suggestion
    // Verify top level folder is lower case and remove empty entries when splitting by slash
```

**Review by mikeharder — APPROVED** [review]

**Review by mikeharder — CHANGES_REQUESTED** [review]
Please add a unit test

**Comment by ckairen** [comment]
> Please add a unit test

This unit test will be built on top of the unit test in this pr https://github.com/Azure/azure-rest-api-specs/pull/26721

blocked until that pr is merged

**Review by mikeharder — APPROVED** [review]

**Pull request merged**